internal/cachecontroller/cache_controller.go

```diff
@@ -127,6 +139,7 @@ func (c *InMemoryCacheController) DetermineInvalidation(
 	_, present := c.inflightInvalidations.LoadOrStore(storeID, struct{}{})
 	if !present {
 		span.SetAttributes(attribute.Bool("check_invalidation", true))
+		checkQueryCacheInvalidationCounter.Inc()
```

Comment:
When placed in this line, this counter is slightly misleading, as you _may_ not end up doing any invalidations at all.

Why not move this line to

```go
for ; idx >= 0; idx-- {
       // here
	t := changes[idx].GetTupleKey()
	c.invalidateIteratorCacheByObjectRelation(storeID, t.GetObject(), t.GetRelation(), lastModified)
	c.invalidateIteratorCacheByObjectTypeRelation(storeID, t.GetUser(), tuple.GetType(t.GetObject()), lastModified)
		}
```

where you know that for sure an invalidation will take place?